### PR TITLE
Add Notion api 

### PR DESCRIPTION
**Pull request by AadityaMunjal**
- [x] My submission is formatted according to the guidelines in the [contributing guide](CONTRIBUTING.md)
- [x] My addition is ordered alphabetically
- [x] My submission has a useful description
- [x] The description does not end with punctuation
- [x] Each table column is padded with one space on either side
- [x] I have searched the repository for any relevant issues or pull requests
- [x] Any category I am creating has the minimum requirement of 3 items
- [x] All changes have been [squashed][squash-link] into a single commit

[squash-link]: <https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -366,6 +366,7 @@ API | Description | Auth | HTTPS | CORS |
 | [Code::Stats](https://codestats.net/api-docs) | Automatic time tracking for programmers | `apiKey` | Yes | No |
 | [File.io](https://www.file.io) | File Sharing | No | Yes | Unknown |
 | [Mercury](https://mercury.postlight.com/web-parser/) | Web parser | `apiKey` | Yes | Unknown |
+| [Notion](https://developers.notion.com/docs/getting-started) | Integrate with Notion | `OAuth` & `apiKey` | Yes | Unknown |
```
Comment (matheusfelipeog) [inline README.md:369]: Use only 1 means of authentication here.

It seems to me that only `OAuth` is enough.

```suggestion
| [Notion](https://developers.notion.com/docs/getting-started) | Integrate with Notion | `OAuth` | Yes | Unknown |
```

**Review by matheusfelipeog — CHANGES_REQUESTED** [review]
Thanks for you contribution @AadityaMunjal 😄 

Can you resolve the review, please?

**Comment by matheusfelipeog** [comment]
After resolving the review, [squash all commits in a single commit](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit), please. ✋

**Comment by AadityaMunjal** [comment]
Hi @matheusfelipeog I've done what you asked for :smile:

**Review by pawelborkar — APPROVED** [review]

**Review by matheusfelipeog — APPROVED** [review]
A big THANKs for updating this PR, @AadityaMunjal 😄

**Pull request merged**